runtime/connectors/blob/blobhandler.go

```diff
@@ -0,0 +1,65 @@
+package blob
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/rilldata/rill/runtime/pkg/fileutil"
+	"gocloud.dev/blob"
+	"golang.org/x/sync/errgroup"
+)
+
+// increasing this limit can increase speed ingestion
+// but may increase bottleneck at duckdb or network/db IO
+// set without any benchamarks
+const concurrentBlobDownloadLimit = 8
```

Comment:
Let's also make this a config parameter, defaulting to 8. Let's call it `GlobMaxConcurrentDownloads`.

Reply:
This seems like a very implementation specific config. I think this shouldn't be exposed to the user.
Also although tough to argue without any benchmarks but we should keep the default high since the go routines are expected to be mostly busy in IO only.

Reply:
I would like this to be configurable since I'm not sure if some users might run into gateway problems (e.g. if there are limits on open connections to same host on single OS, or someone uses an S3-compatible data lake that has rate limits, etc.). In support, it might be handy to have people try reduce this to 1.

Reply:
> > > runtime/connectors/blob/blobdownloader.go
> > 
> > 
> > This sounds good. However I would definitely want to give downloadAll and downloadSingle option to clients.
> 
> The only clients are the S3 and GCS connectors – and they configure by either passing a glob or non-glob path. They know that only a single download happens if they don't pass a glob.
> 
> It's better to have minimal public interfaces and keep internals private. If a future use case needs more granular access to the internals, then we can refactor private functions to public functions, but not before :)

As I said earlier downloading all files at one shot may not be a very good impl. 
What if we want to just download a subset of files for some olap system and download all for others. There could be other use cases as well which may not need all files. To implement this we need to expose it to either driver or some orchestrator. 
Also on second thought I would not have downloadAll functionality and let clients download in parallel if they want.